google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/spi/v2/HttpBigQueryRpc.java

```diff
@@ -515,6 +528,26 @@ public TableDataList listTableData(
     }
   }
 
+  @Override
+  public TableDataList listTableDataWithRowLimit(
```

Comment:
Is this supposed to be in the same PR?

Reply:
Yes, we are adding this new RPC call to get data using `tabledata.list` API with a "rowLimit" which specifies how many rows to "pre-fetch".